.github/workflows/tools.yml

```diff
@@ -0,0 +1,42 @@
+# Test tools directories that have their own go.mod files, and
+# otherwise wouldn't get tested along with the main go-algorand build.
+name: Test tools modules
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'tools/block-generator/**'
+      - 'tools/x-repo-types/**'
+  pull_request:
+    paths:
+      - 'tools/block-generator/**'
+      - 'tools/x-repo-types/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+      - name: Install libraries
+        run: sudo apt-get -y -q install libboost-math-dev
+      # move go out of the way temporarily to avoid "go list ./..." from installing modules
+      - name: Make libsodium.a
+        run: sudo mv /usr/bin/go /usr/bin/go.bak && make crypto/libs/linux/amd64/lib/libsodium.a && sudo mv /usr/bin/go.bak /usr/bin/go
+      - name: Add bin to PATH
+        run: |
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+          echo "$RUNNER_WORKSPACE/$(basename $GITHUB_REPOSITORY)/bin" >> $GITHUB_PATH
+      - name: Install specific golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.17.13'
```

Comment:
Do you know if there's a way to get this dynamically from our `get_golang_version.sh` script. No worries, if not. I can investigate.